templates/webapi/main/parent_group_overview.html.ep

```diff
@@ -15,27 +15,131 @@
+      removeClasses("#grouped_by_build_tab", classesToRemove);
+      $("#grouped_by_group").addClass("active show");
+      $("#grouped_by_group_tab").addClass(" parent_group_overview_grouping_active active");
+    }
+  }
+
+  $(document).ready(function(){
+    var hash = window.location.hash
+
+    if (hash === undefined || hash === "")
+      hash = $.cookie(cookie_product_name);
+
+    if (hash === undefined || hash === "")
+      hash = $.cookie(cookie_all_products_name);
+
+    if (hash === undefined || hash === "")
+      % if ($group->{build_view_mode} == 0) {
+        hash = "#grouped_by_build";
+      % } else {
+        hash = "#grouped_by_group";
+      % }
+
+    if (hash === undefined || hash === "")
+      hash = "#grouped_by_build";
+
+    window.location.hash = hash;
+
+    update_grouped_by_classes();
+    update_grouped_by_default_links();
+
+    $(".nav-tabs a").click(function(){
+      var hash = $(this).attr('href');
+      window.location.hash = hash;
+      update_grouped_by_default_links();
+      update_grouped_by_classes();
+    })
+
+    $("#grouped_by_default").click(function(){
+      $.cookie(cookie_product_name, window.location.hash);
+      update_grouped_by_default_links();
+    })
+
+    $("#grouped_by_default_all").click(function(){
+      $.cookie(cookie_all_products_name, window.location.hash);
+      update_grouped_by_default_links();
+    })
+  })
+</script>
+
+<h2 class="mb-4">
     Last Builds for <%= $group->{name} %>
 </h2>
+
+%= include 'main/pinned_comments'
+
 <div class="well well-lg" id="group_description">
     % if (my $description = $group->{rendered_description}) {
         %= $description
     % }
-    <div id="child_groups">
-        <h3>Accumulated results for</h3>
-        <ul>
-        % for my $child_group (@$child_groups) {
-            <li>
-                %= link_to $child_group->name => url_for('group_overview', groupid => $child_group->id)
-            </l>
-        % }
-        </ul>
-    </div>
 </div>
-%= include 'main/pinned_comments'
 
 <div id="build-results"></div>
-%= include 'main/group_builds', build_results => $build_results, group => $group, children => $children, default_expanded => 1
+
+<div class="tab-content" id="grouped_by_tabContent">
+  <div class="tab-pane fade" id="grouped_by_build" role="tabpanel" aria-labelledby="grouped_by_build_tab">
+    %= include 'main/group_builds', build_results => $build_results, group => $group, children => $children, default_expanded => 1
+  </div>
+  <div class="tab-pane fade" id="grouped_by_group" role="tabpanel" aria-labelledby="grouped_by_group_tab">
+    %= include 'main/group_builds_functionality_view', build_results => $build_results, group => $group, children => $children, default_expanded => 1
+  </div>
+</div>
+
 %= include 'main/more_builds', limit_builds => $limit_builds
+
+
+<div class="d-flex flex-row bd-highlight mb-3">
+  <div class="bd-highlight">
+    <ul class="nav nav-pills">
+      Grouped
```

Comment:
I'm wondering whether this is valid HTML. Shouldn't the ul element only contain li elements and not a raw text node?

Reply:
Yes, but is a trick. I need the ul to use bootstrap pills and at the same time I need to keep the visual consistency. It works but it is a problem I could try to fix it

Reply:
I've just checked and the HTML validator says no. So this should definitely be achieved differently then. Having valid HTML is more important than doing it "the framework's way".

Reply:
Fixed